pkg/operation/botanist/addons.go

```diff
@@ -382,7 +382,12 @@ func (b *Botanist) generateCoreAddonsChart() (*chartrenderer.RenderedChart, erro
 		nodeExporterConfig        = map[string]interface{}{}
 		blackboxExporterConfig    = map[string]interface{}{}
 		nodeProblemDetectorConfig = map[string]interface{}{}
-		networkPolicyConfig       = map[string]interface{}{}
+		networkPolicyConfig       = map[string]interface{}{
+			"nodeLocalDNS": map[string]interface{}{
+				"enabled": b.Shoot.NodeLocalDNSEnabled,
+				"ip":      b.Shoot.Networks.CoreDNS.String(),
```

Comment:
Maybe rename it to make it more clear.
```suggestion
				"kubeDNSClusterIP":      b.Shoot.Networks.CoreDNS.String(),
```